base/shared/src/main/scala/scalaz/data/ilist.scala

```diff
@@ -316,6 +317,9 @@ private[data] object IListImpl extends IListModule {
   def cons[A](a: A, as: Fix[Maybe2[A, ?]]): IList[A] =
     Fix.fix[Maybe2[A, ?]](Maybe2.just2(a, as))
 
+  def delay[A](as: () => IList[A]): IList[A] =
+    Fix.fix[Maybe2[A, ?]](Maybe2.delay(() => uncons(as()))) // TODO: use subst?
```

Comment:
If we're gonna go down the non-parametric `Lazy` route, then yeah it would be nice to provide something like `subst` here. I've been writing `Coercible` but it's stalled lately, I think that would be optimal, but a hacky `subst` defn or `Is` value for now is fine.